crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -525,3 +533,144 @@ fn test_verify_nonce_in_unsigned_tx() {
+            transaction_revert_class
+        );
+
+        let salt = Felt252Wrapper::ZERO;
+
+        let deploy_transaction = InvokeTransactionV1 {
+            sender_address: account_addr.into(),
+            signature: vec![],
+            nonce: Felt252Wrapper::ONE,
+            calldata: vec![
+                Felt252Wrapper::ONE,
+                Felt252Wrapper::from_hex_be(UDC_ADDRESS).unwrap(), // udc address
+                Felt252Wrapper::from_hex_be(UDC_SELECTOR).unwrap(), // deployContract selector
+                Felt252Wrapper::from_hex_be("0x4").unwrap(),       // calldata len
+                transaction_revert_class_hash,                     // contract class hash
+                salt,                                              // salt
+                Felt252Wrapper::ONE,                               // unique
+                Felt252Wrapper::ZERO,                              // constructor calldata len
+            ],
+            max_fee: u128::MAX,
+            offset_version: false,
+        };
+
+        // validate invoke transaction
+        assert_ok!(Starknet::validate_unsigned(
+            TransactionSource::InBlock,
+            &crate::Call::invoke { transaction: deploy_transaction.clone().into() },
+        ));
+
+        let contract_address: FieldElement = get_udc_deployed_contract_address(
+            salt.into(),
+            transaction_revert_class_hash.into(),
+            &UdcUniqueness::Unique(UdcUniqueSettings {
+                deployer_address: FieldElement::from_hex_be(
+                    "0x0642a8b9e2c6cc3a9ddb84575123f262a21415f78db453b0625d889e1e06ac32",
+                )
+                .unwrap(),
+                udc_contract_address: FieldElement::from_hex_be(UDC_ADDRESS).unwrap(),
+            }),
+            &[],
+        );
+
+        // deploy contract
+        assert_ok!(Starknet::invoke(RuntimeOrigin::none(), deploy_transaction.into()));
+
+        // create increase balance transaction
+        let increase_balance_tx = InvokeTransactionV1 {
+            sender_address: account_addr.into(),
+            signature: vec![],
+            nonce: Felt252Wrapper::TWO,
+            max_fee: u128::MAX,
+            offset_version: false,
+            calldata: vec![
+                Felt252Wrapper::ONE,
+                contract_address.into(),
+                Felt252Wrapper::from_hex_be(INCREASE_BALANCE_FUNCTION_SELECTOR).unwrap(),
+                Felt252Wrapper::from_hex_be("0x1").unwrap(),
+                Felt252Wrapper::from_hex_be("0xa").unwrap(),
+            ],
+        };
+
+        // the transaction reverts and returns Ok
+        assert_ok!(Starknet::invoke(RuntimeOrigin::none(), increase_balance_tx.clone().into()));
+
+        let storage_var_selector = StorageKey(PatriciaKey(
+            StarkFelt::try_from("0x0206f38f7e4f15e87567361213c28f235cccdaa1d7fd34c9db1dfe9489c6a091").unwrap(),
```

Comment:
can we call the get_balance method instead so we're sure we're reading it correctly? because if the storage key is some random hex it will still return 0. also, as far as I remember, this bug exists in Madara today so this test case should actually fail.

Reply:
Sure, will call the get_balance method. But the storage key is not just random hex -- it is the variable selector. The hex translates to `balance` in string. 

For better assertion, will add the get_balance method

Reply:
Yes this test actually fails for current madara implementation.

Reply:
yet to add the get_balance function call. Pushing the changes in sometime

Reply:
This is done